pkg/distributor/distributor.go

```diff
@@ -506,6 +506,15 @@ func removeLabel(labelName string, labels *[]mimirpb.LabelAdapter) {
 	}
 }
 
+// Remove labels with value=="" from a slice of LabelPairs, updating the slice in-place.
+func removeEmptyLabelValues(labels *[]mimirpb.LabelAdapter) {
+	for i := len(*labels) - 1; i >= 0; i-- {
```

Comment:
Could there be any performance difference when processing labels from the beginning vs from the end?

Reply:
I suspect the idea was if we have `a, b, c, d, e` and we're going to remove `b` and `d`, then going forwards means we copy `d` before removing it, whereas going backwards avoids that.

Also, going forward you need to not increment `i` if you removed an item.